src/realm/sync/config.hpp

```diff
@@ -45,9 +45,11 @@ using port_type = std::uint_fast16_t;
 enum class ProtocolError;
 } // namespace sync
 
-struct SyncError : public SystemError {
+struct SyncError {
     enum class ClientResetModeAllowed { DoNotClientReset, RecoveryPermitted, RecoveryNotPermitted };
 
+    Status status;
```

Comment:
Status should probably be a `const`, since there are two `string_view`s that depend on its contents.

Reply:
Making this const makes SyncError non-assignable - i.e. you can't do something like `SyncError copy = other_sync_error` or `std::optional<SyncError> maybe_sync_error; maybe_sync_error = sync_error;` I'm not sure adding that restriction to this type is worth it compared to say a comment saying not to move out the Status if you intend to re-use the SyncError?

Reply:
Understood - the side effects of `const` are not worth the extra headaches... and a `const status()` accessor is less efficient.
A comment would probably be good, since one may not notice the `string_view` references without looking into the class implementation.